orchestrator/cli/generate.py

```diff
@@ -11,56 +11,67 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import os
+import sys
 from pathlib import Path
 from typing import Dict, Optional
 
+import structlog
 import typer
 import yaml
 from jinja2 import Environment, FileSystemLoader
 
+from orchestrator.cli.generator.generator.helpers import get_variable
 from orchestrator.cli.generator.generator.migration import generate_product_migration
 from orchestrator.cli.generator.generator.product import generate_product
 from orchestrator.cli.generator.generator.product_block import generate_product_blocks
 from orchestrator.cli.generator.generator.settings import product_generator_settings as settings
 from orchestrator.cli.generator.generator.unittest import generate_unit_tests
 from orchestrator.cli.generator.generator.workflow import generate_workflows
 
+logger = structlog.getLogger(__name__)
+
 app: typer.Typer = typer.Typer()
 
 
-def read_config(config_file: str) -> Optional[Dict]:
+def read_config(config_file: Path) -> Dict:
     try:
         with open(config_file) as stream:
             try:
                 return yaml.safe_load(stream)
             except yaml.YAMLError:
-                typer.echo("Failed to parse configuration file.")
+                logger.error("failed to parse configuration file", config_file=str(config_file))
     except FileNotFoundError:
-        typer.echo(f'File "{config_file}" not found')
+        logger.error("configuration file not found", config_file=str(config_file))
 
-    return None
+    # when we cannot get the product configuration somehow, it does not make sense to continue ...
+    sys.exit(1)
```

Comment:
True!

It would be better to raise a (perhaps custom) exception though, as sys.exit will attempt to exit the current interpreter, which may not always be what you want.

Reply:
Good point. I removed the exit(1) and now just re-raise the exception after logging the problem.